typing_inspect.py

```diff
@@ -116,6 +116,24 @@ class MyClass(Tuple[str, int]):
     return type(tp) is TupleMeta
 
 
+def is_optional_type(tp):
+    """Returns `True` if the type is `type(None)`, or is a direct `Union` to `type(None)`, such as `Optional[T]`.
+
+    WARNING: this method might return a misleading `False` if
+     - `tp` is a `TypeVar` bound, or constrained to, an optional type
+     - `tp` is a `Union` to a `TypeVar` bound or constrained to an optional type,
+     - `tp` refers to a *nested* `Union` containing an optional type or one of the above.
+    """
+
+    if tp is type(None):
+        return True
+    elif is_union_type(tp):
+        # TODO should we force evaluate to be True here or set it as a parameter?
```

Comment:
I think it is fine to force it. This is what most people expect.

Reply:
ok